src/onegov/org/views/settings.py

```diff
@@ -302,3 +312,78 @@ def truncate(text):
     icon='fa-calendar-alt', order=-700)
 def handle_event_settings(self, request, form, layout=None):
     return handle_generic_settings(self, request, form, _("Events"), layout)
+
+
+@OrgApp.form(
+    model=Organisation, name='api-keys', template='api_keys.pt',
+    permission=Secret, form=OneGovApiSettingsForm, setting=_("OneGov API"),
+    icon='fa-user-lock', order=1)
+def handle_api_keys(self: Organisation, request: CoreRequest,
+                    form: OneGovApiSettingsForm, layout=None):
+    """Handles the generation of API access keys."""
+
+    collection = UserCollection(request.session)
+    user = collection.by_username(request.identity.userid)
+    if not user:
+        raise HTTPForbidden()
+
+    if form.submitted(request):
+        key = ApiKey(
+            name=form.name.data,
+            read_only=True,
+            last_used=None,
+            key=uuid4(),
+            user=user,
+        )
+        request.session.add(key)
+        request.session.flush()
+        request.success(_("Your changes were saved"))
+
+    layout = layout or SettingsLayout(self, request)
+
+    def current_api_keys_by_user(
+        request: CoreRequest, self: Organisation, user: User, layout
+    ):
+        # TODO: how to make mypy happy for `api_keys` backref?
```

Comment:
You can forward declare the backref on the user Model inside a `if TYPE_CHECKING` block i.e.
```python
if TYPE_CHECKING:
    api_keys: relationship[list[ApiKey]]
```
It's possible that `test_hierarchy` will complain if you do that, since an import statement will create a soft dependency even while inside an `if TYPE_CHECKING:` block. You might be able to get around it by using `importlib` instead. Alternatively you can punt and annotate it as `relationship[list[Any]]`.